File: src/ShitLeopard.DataLayer/Entities/ScriptLine.cs
using System;
using System.Collections.Generic;

namespace ShitLeopard.DataLayer.Entities
{
    public partial class ScriptLine
    {
        public ScriptLine()
        {
            ScriptWord = new HashSet<ScriptWord>();
        }

        public long Id { get; set; }
        public string Body { get; set; }
        public long ScriptId { get; set; }
        public long? CharacterId { get; set; }

        public virtual Character Character { get; set; }
        public virtual Script Script { get; set; }
        public virtual ICollection<ScriptWord> ScriptWord { get; set; }
    }
}

File: src/ShitLeopard.Api/src/viewModels/ScriptLine.ts
import { Character } from './Character';
import { Script } from './Script';
import { ScriptWord } from './ScriptWord';

export class ScriptLine {
  constructor(
    public id?: number,
    public body?: string,
    public scriptId?: number,
    public characterId?: number,
    public character?: Character,
    public script?: Script,
    public scriptWord?: ScriptWord[],
    public episodeId?: number,
    public seasonId?: number,
    public episodeTitle?: string
  ) {}
}
